nautobot_golden_config/utilities/helper.py

```diff
@@ -63,17 +64,26 @@ def verify_global_settings(logger, global_settings, attrs):
             raise NornirNautobotException()
 
 
-def check_jinja_template(obj, logger, template):
-    """Helper function to catch Jinja based issues and raise with proper NornirException."""
+def render_jinja_template(obj, logger, template):
+    """
+    Helper function to render Jinja templates.
+
+    Args:
+        obj (Device): The Device object from Nautobot.
+        logger (NornirLogger): Logger to log error messages to.
+        template (str): A Jinja2 template to be rendered.
+
+    Returns:
+        str: The ``template`` rendered.
+
+    Raises:
+        NornirNautobotException: When there is an error rendering the ``template``.
+    """
     try:
-        template_rendered = Template(template, undefined=StrictUndefined).render(obj=obj)
-        return template_rendered
+        return render_jinja2(template_code=template, context={"obj": obj})
     except UndefinedError as error:
         logger.log_failure(obj, f"Jinja `{template}` has an error of `{error}`.")
         raise NornirNautobotException()
-    except TemplateSyntaxError as error:
```

Comment:
this error can't happen anymore?

Reply:
They all stem from TemplateError, and we are trying to catch all TemplateErrors, so we can just use the base Exception Class

Reply:
I agree that there is no functional difference right now, but if we are going to update this, i think we should update the error message to be more clear to the type of error that there is.